### PR TITLE
Update and clean up .gitignore

### PR DESCRIPTION
**Pull request by uklotzde**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,12 +7,7 @@
 *.log
 *.log.*
 
-# Compilation artifacts
-src/mixxx.rc.include
-src/mixxx.res
-src/test/**/*.actual
-res/qrc_mixxx.cc
-
+# TODO: Do we really need this exclude?
```
Comment (Holzhaus) [inline .gitignore:10]: Yes, this may be created if you trun a python tool from the tools directory.

Comment (uklotzde) [reply .gitignore:10]: Ok, I'll modify this commit and force push.

**Comment by JoergAtGithub** [comment]
.vs is generated with every build on Windows, but no file in this directory should go in version control, like pre complied headers, build configuration

**Comment by uklotzde** [comment]
> .vs is generated with every build on Windows, but no file in this directory should go in version control, like pre complied headers, build configuration

Please add this top your global .gitignore as the top-level comment states. We will not add excludes for every possible IDE out there.

**Comment by JoergAtGithub** [comment]
This is not only IDE, but the each Mixxx build generate them. Everybody on Windows is affected!

**Comment by Holzhaus** [comment]
I'm pretty sure I didn't see this when running CMake from the command line on windows. Are you sure?

**Comment by JoergAtGithub** [comment]
If you build with pre-compiled headers, they are stored in this directory

**Comment by uklotzde** [comment]
> If you build with pre-compiled headers, they are stored in this directory

But this directory is generated by the IDE, not by CMake. otherwise

> If you build with pre-compiled headers, they are stored in this directory

Command-line or IDE? IDE builds are your own responsibility. All CMake build artifacts are already excluded by /build directory.

**Comment by JoergAtGithub** [comment]
It's build from IDE without any settings from another source than CMake. It's nothing custom in my environment. This folder appears for every developer who builds locally following the Mixxx build instructions. And it doesn't contain anything that should go into version control.
Please don't break working things!

**Comment by uklotzde** [comment]
@JoergAtGithub As expected, you confirmed that it is generated by Visual Studio. The same applies to .vscode and .idea folders generated by other IDEs. Those excludes belong into your global .gitignore and are not supposed to be repeated in every project.

I am aware that many projects are doing this wrong and cluttering their .gitignore even with .DS_Store entries.

**Comment by Holzhaus** [comment]
@JoergAtGithub https://openqa-bites.github.io/posts/2021-04-23-global-gitignore/

**Review by Holzhaus — APPROVED** [review]

**Comment by uklotzde** [comment]
Ping

**Pull request merged**